src/main/java/org/dcsa/ebl/controller/ShippingInstructionController.java

```diff
@@ -24,12 +25,18 @@
 
   @GetMapping(path = "/{shippingInstructionID}")
   public Mono<ShippingInstructionTO> findById(@PathVariable String shippingInstructionID) {
-    return shippingInstructionService.findById(shippingInstructionID);
+    return shippingInstructionService
+        .findById(shippingInstructionID)
+        .switchIfEmpty(
+            Mono.error(
+                ConcreteRequestErrorMessageException.notFound(
+                    "No Shipping Instruction found with ID: " + shippingInstructionID)));
```

Comment:
I would move this to the service (just after the repository call).  It is fine if the controller **also** have this check but it should be an internal server error because the service should never return an empty mono.

Reply:
Done